src/lib/AssetConnection.ts

```diff
@@ -77,8 +77,8 @@ class AssetConnection {
         resolution: this.resolution,
       })
       .catch((reason) => {
-        this.logger.logError(`Error entering channel presence: ${reason}`);
-        throw new Error(reason);
+        this.logger.logError(`Error entering channel presence: ${JSON.stringify(reason)}`);
```

Comment:
I actually think `reason` or `reason.message` would be better here. The reason will be an ably-js errorinfo which has a `toString` convenience method which will look a bit nicer than the JSON string. I think this is probably the best fix anyway:
```suggestion
        this.logger.logError(`Error entering channel presence: ${(e as AblyTypes.ErrorInfo).message}`);
```

Reply:
Oh nice one! Updated 😄

Reply:
sorry forgot to mention - the same applies to the other JSON.stringify you added 😄

Reply:
of course 🤦 done!